include/sisl/fds/buffer.hpp

```diff
@@ -74,6 +74,19 @@ struct sg_iterator {
         return ret_iovs;
     }
 
+    void move_offset(uint32_t const size) {
+        uint64_t remain_size = size;
+
+        for (; (remain_size > 0) && (m_cur_index < m_input_iovs.size()); ++m_cur_index, m_cur_offset = 0) {
```

Comment:
Should cache `m_input_iovs.size()` so this expression is cheap.

Reply:
Is there a significant overhead for getting the size of a vector?

Reply:
No, but it's like doing `i++;` instead of `++i` when the temporary is not needed. Just extra cpu steps for each iteration that aren't used. Logically correct.